### PR TITLE
Add DrawPolygons to Worldview

### PR DESCRIPTION
**Pull request by vidaaudrey**
## Summary
Add the `DrawPolygon` created by @brianc.  We have been using this command internally, and open sourcing it will help add polygon drawing capability to the 3D panel in Webviz. 

Note: use with caution as there might be major api changes before we completely open source it with examples and documentations.

## Test plan
Unit test and manual test.

## Versioning impact
Will publish a patch version now and a patch version later (after documentation is added).

### CODE REVIEW

File: packages/regl-worldview/src/commands/DrawPolygon/PolygonBuilder.js [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+  return x1 === x2 && y1 === y2 && z1 === z2;
+}
+
+function isClosed(polygon: Polygon): boolean {
+  const { points } = polygon;
+  for (let i = 0; i < points.length - 1; i++) {
+    if (areEqual(points[i], points[i + 1])) {
+      return true;
+    }
+  }
```
Comment (janpaul123) [inline packages/regl-worldview/src/commands/DrawPolygon/PolygonBuilder.js]: Huh this for loop seems strange to me. If you create a polygon with two points that sit right on top of each other for some reason we consider it closed?

File: packages/regl-worldview/src/commands/DrawPolygon/PolygonBuilder.js [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+  return areEqual(firstPoint, lastPoint);
+}
+
+// has listeners you can pass to Worldview for mouse interactions
+// internally builds a list of polygons and modifies the polygons
+// based on mouse & keyboard interactions.  for now we use mututation internally
+// instead of immutability to keep the number of allocations lower and make
+// the implementation a bit more straightforward
+//
+// for an example implementation check stories/Worldview/drawPolygon.js
```
Comment (janpaul123) [inline packages/regl-worldview/src/commands/DrawPolygon/PolygonBuilder.js]: Out of date?

File: packages/regl-worldview/src/commands/DrawPolygon/PolygonBuilder.js [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+export default class PolygonBuilder {
+  mouseDown: boolean = false;
+  polygons: Polygon[];
+  onChange: () => void = () => {};
+  activePolygon: ?Polygon;
+  activePoint: ?PolygonPoint;
+  mouseDownPoint: Vec3;
```
Comment (janpaul123) [inline packages/regl-worldview/src/commands/DrawPolygon/PolygonBuilder.js]: Might be good to prefix private variables and methods with an underscore.

File: packages/regl-worldview/src/commands/DrawPolygon/PolygonBuilder.js (hunk 1 above)
Comment (vidaaudrey) [reply packages/regl-worldview/src/commands/DrawPolygon/PolygonBuilder.js]: Not sure what was the original thought. I'll create a ticket for the after-work and add to the todo list.

File: packages/regl-worldview/src/commands/DrawPolygon/PolygonBuilder.js (hunk 3 above)
Comment (vidaaudrey) [reply packages/regl-worldview/src/commands/DrawPolygon/PolygonBuilder.js]: `mouseDownPoint` and other instance variable seem to be useful for public api as well, e.g. trigger certain external events based mouseDown position

File: packages/regl-worldview/src/commands/DrawPolygon/PolygonBuilder.js (hunk 2 above)
Comment (vidaaudrey) [reply packages/regl-worldview/src/commands/DrawPolygon/PolygonBuilder.js]: Good catch! Removed.

**Pull request merged**